cli/azd/cmd/init.go

```diff
@@ -188,13 +189,21 @@ func (i *initAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	switch initTypeSelect {
 	case initAppTemplate:
 		tracing.SetUsageAttributes(fields.InitMethod.String("template"))
-		err := i.initializeTemplate(ctx, azdCtx)
+		template, err := i.initializeTemplate(ctx, azdCtx)
 		if err != nil {
 			return nil, err
 		}
 
-		err = i.initializeEnv(ctx, azdCtx)
-		if err != nil {
+		var templateMetadata *templates.Metadata
+		if template != nil {
+			templateMetadata = &template.Metadata
+		}
+
+		if err := i.initializeEnv(ctx, azdCtx, templateMetadata); err != nil {
+			return nil, err
+		}
+
+		if err := i.initializeProject(ctx, azdCtx, templateMetadata); err != nil {
```

Comment:
Can you help me understand what we are doing here?  
When we run `template, err := i.initializeTemplate(ctx, azdCtx)` , the template we received contains metadata that we need to write to the `azure.yaml` right?  (that's probably the case where the user picked an ADE interactively, so there's not an existing azure.yaml).  

If that's the case, `initializeProject()` should be part of `initializeTemplate()`, instead of having it as a last step (after azure.yaml and .env have been already been written).

Consider moving this `initializeProject` logic to be part of the `repository/initializer.go`, as that's where we are writing the `projectConfig`.  We already have a `repoInitializer.InitializeMinimal` and `repoInitializer.Initialize`.  We might want to have a `repoInitializer.InitializeMetadata`

Reply:
I'll take a look.

Reply:
Move initializing project (azure.yaml) updates to repo initializer where this does make sense. Thanks for the suggestion.

Yes - think of this is a dynamic template that contains additional metadata of configuration changes that should get applied to the projects `azure.yaml` after repo is cloned.